packages/snap/src/keyring.ts

```diff
@@ -308,22 +314,31 @@ export class AccountAbstractionKeyring implements Keyring {
   ): Promise<SubmitRequestResponse> {
     const { method, params = [] } = request.request as JsonRpcRequest;
 
-    if (method === InternalMethod.SetConfig) {
-      return {
-        pending: false,
-        result: await this.setConfig((params as [ChainConfig])[0]),
-      };
+    switch (method) {
+      case InternalMethod.GetConfigs: {
+        return {
+          pending: false,
+          result: await this.getConfigs(),
+        };
+      }
+      case InternalMethod.SetConfig: {
+        return {
+          pending: false,
+          result: await this.setConfig((params as [ChainConfig])[0]),
```

Comment:
The `[0]` shouldn't be needed since `.setConfig` already return the config of the given `chainId` no?

Reply:
The SetConfig passes only a specific chainconfig, and the params are an array of arguments. We are just taking the first element.

Reply:
my bad... I misread the expression, I thought we were indexing the returned value of `.setConfig`, but we actually are indexing `params` (considered to be of type `[ChaindConfig]`)! 🙈